web/src/message_edit.ts

```diff
@@ -452,6 +452,37 @@ function handle_inline_topic_edit_keydown(this: HTMLElement, e: JQuery.KeyDownEv
 
 function handle_inline_topic_edit_change(this: HTMLInputElement): void {
     const $inline_topic_edit_input = $(this);
+
+    // We use a hidden span element, which we update with the value
+    // of the input field on every input change to calculate the
+    // width of the topic value. This allows us to dynamically adjust
+    // the max-width of the input field.
+    const $topic_value_mirror = $inline_topic_edit_input
+        .closest(".topic_edit_form")
+        .find(".topic_value_mirror");
+    $topic_value_mirror.text(this.value);
+    let topic_width = $topic_value_mirror.width();
+    if (this.value.length > 0) {
+        // When the user starts typing in the inline topic edit input field,
+        // we dynamically adjust the max-width of the input field to match
+        // width of the text in the input field + 1ch width for some cushion.
+        $inline_topic_edit_input.css("max-width", `calc(${topic_width}px + 1ch)`);
+    } else {
+        // When the user deletes all the text in the inline topic edit input field,
+        // we check if the input field has a placeholder and if it does, we set the
+        // max-width of the input field to the length of the placeholder + 1ch
+        // width for some cushion.
+        const placeholder = $inline_topic_edit_input.attr("placeholder");
+        if (placeholder !== undefined) {
+            $topic_value_mirror.text(placeholder);
+            topic_width = $topic_value_mirror.width();
+            $inline_topic_edit_input.css("max-width", `calc(${topic_width}px + 1ch)`);
+        } else {
+            // Otherwise, we set the max-width of the input field to the max_topic_length.
+            $inline_topic_edit_input.css("max-width", realm.max_topic_length + "ch");
+        }
```

Comment:
This case seems a bit wrong -- we should just need the width for the placeholder, not the maximum possible width.

![image](https://github.com/user-attachments/assets/cf4deff6-327f-40db-9410-03c6b2dd260a)

Reply:
After discussion at CZO, set the (empty input + no placeholder) width to 20ch,